src/rollback_to_stable/rts_btree.c

```diff
@@ -735,8 +741,8 @@ __rts_btree_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
          * windows can be the same. To abort these updates, check for any stable update from history
          * store or remove the key.
          */
-        if (tw->start_ts == tw->stop_ts && tw->durable_start_ts == tw->durable_stop_ts &&
-          tw->start_txn == tw->stop_txn) {
+        if (WT_TIME_WINDOW_HAS_STOP_PREPARE(tw) && tw->start_prepared_id == tw->stop_prepared_id &&
```

Comment:
Here I think it is wrong. In the previous check, it covers both the prepared and non-prepared case. Now it only covers the prepared case.

Reply:
hm, I'll change it to
`if (tw->start_prepared_id == tw->stop_prepared_id &&
          tw->start_prepare_ts == tw->stop_prepare_ts && tw->start_txn == tw->stop_txn && tw->start_ts == tw->stop_ts && tw->durable_start_ts == tw->durable_stop_ts)`, does that sound good to you>

Reply:
done

Reply:
actually after i changed this logic, `test_rollback_to_stable19` failed ([link](https://parsley.mongodb.com/evergreen/wiredtiger_ubuntu2004_unit_test_bucket01_patch_a06f972f450b20a556e1983e28ad44e7d57c8127_687de0c529d5cb0007ce93f4_25_07_21_06_40_06/0/task?bookmarks=0,2332,3476)) so I'm changing it back to the first version for now